### PR TITLE
Add FXIOS-7901 [v124] Add menu controller options in WebEngine

### PR DESCRIPTION
**Pull request by lmarceau**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7901)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/17646)

## :bulb: Description
So we have this thing called `MenuHelper` in Firefox iOS, where it's used in different parts of the application to add menu controller options (see screenshots in the JIRA for more info about what those options are). There's mainly three areas we have options added: login, webview and url bar. Now the goal of this task was to add the possibility to add webview textfield menu options in the `WebEngine`, but I don't want to have the forced option to have all menu options added. To avoid this, the `MenuHelperInterface` is splitted in three sub-protocols. One for the URL bar, one for the logins and one for the web view. This will enable Focus to have the Webview menu options without having to implement the URL bar and logins one. 

The new `MenuHelper` is in `Common` so the `WebEngine` can use it. This means we inject the localized strings from the Client.

Here's the implementation in the `SampleBrowser`:

https://github.com/mozilla-mobile/firefox-ios/assets/11338480/1a6daf0b-0e35-4081-b9d3-85134f4d4ed1

## :pencil: Checklist
You have to check all boxes before merging
- [X] Filled in the above information (tickets numbers and description of your work)
- [X] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [X] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: BrowserKit/Tests/WebEngineTests/WKEngineSessionTests.swift [hunk 1]
```diff
@@ -403,6 +403,32 @@ final class WKEngineSessionTests: XCTestCase {
...
+        let subject = createSubject()
+        subject?.delegate = engineSessionDelegate
+        let expectedSelection = "A search"
+
+        subject?.tabWebView(webViewProvider.webView, findInPageSelection: expectedSelection)
+
+        XCTAssertEqual(engineSessionDelegate.findInPageCalled, 1)
+        XCTAssertEqual(engineSessionDelegate.savedFindInPageSelection, expectedSelection)
+        prepareForTearDown(subject!)
+    }
```
Comment (mattreaganmozilla) [inline BrowserKit/Tests/WebEngineTests/WKEngineSessionTests.swift:418]: Do these tests actually exercise any code running in the client? It feels like in practice some of our mocks right now only test their own artificial code that runs only on CI. (This is a broader topic not specific just to this PR but I'm wondering if it's something that would be worth us revisiting at some point.)

File: firefox-ios/Client/TabManagement/Tab.swift [hunk 2]
```diff
@@ -1052,7 +1052,7 @@ class TabWebView: WKWebView, MenuHelperInterface, ThemeApplicable {
     }
 
     override func canPerformAction(_ action: Selector, withSender sender: Any?) -> Bool {
-        return super.canPerformAction(action, withSender: sender) || action == MenuHelper.SelectorFindInPage
+        return super.canPerformAction(action, withSender: sender) || action == MenuHelperWebViewModel.selectorFindInPage
```
Comment (mattreaganmozilla) [inline firefox-ios/Client/TabManagement/Tab.swift:1055]: Can you clarify why the explicit check for the `selectorFindInPage` is needed?

**Review by mattreaganmozilla — APPROVED** [review]
👍

File: BrowserKit/Tests/WebEngineTests/WKEngineSessionTests.swift (hunk 1 above)
Comment (lmarceau) [reply BrowserKit/Tests/WebEngineTests/WKEngineSessionTests.swift:418]: In the current case, the real object `WKEngineSession` get tested, but everything around it are Mocks so we can control the output of the test. `WKEngineSession` is not used in any parts under the client app as of now thought. Not sure if this answers the question you asked?

File: firefox-ios/Client/TabManagement/Tab.swift (hunk 2 above)
Comment (lmarceau) [reply firefox-ios/Client/TabManagement/Tab.swift:1055]: Let me look 👀

File: BrowserKit/Tests/WebEngineTests/WKEngineSessionTests.swift (hunk 1 above)
Comment (mattreaganmozilla) [reply BrowserKit/Tests/WebEngineTests/WKEngineSessionTests.swift:418]: Ok thanks @lmarceau for clarifying 👍 I was just wondering how much value we get out of some of our mocks running on CI (again, not specific to this PR, more of a general question). LGTM

**Pull request merged**